File: frontend/lib/L1_domain/entities/goals/remote_tracker.dart
// Copyright (c) 2022. Alexandr Moroz

import '../auth/workspace.dart';
import '../base_entity.dart';

class RemoteTrackerType extends Titleable {
  RemoteTrackerType({required int id, required String title}) : super(id: id, title: title);
}

class RemoteTracker extends RPersistable {
  RemoteTracker({
    required int id,
    required this.type,
    required this.url,
    required this.loginKey,
    required this.description,
    required this.workspace,
    this.connected = false,
  }) : super(id: id);

  final RemoteTrackerType type;
  final String url;
  final String loginKey;
  final String? description;
  final bool connected;
  final Workspace workspace;

  RemoteTracker copyWithConnected(bool _connected) => RemoteTracker(
        id: id,
        type: type,
        url: url,
        loginKey: loginKey,
        description: description,
        workspace: workspace,
        connected: _connected,
      );
}

File: backend/lib/L1_domain/entities/goals/remote_tracker.py
#  Copyright (c) 2022. Alexandr Moroz

from dataclasses import dataclass

from ..auth.workspace_bounded import WorkspaceBounded
from ..base_entity import Persistable, Titleable


@dataclass
class RemoteTrackerType(Titleable):
    pass


@dataclass
class RemoteTracker(Persistable, WorkspaceBounded):

    type: RemoteTrackerType | None = None
    url: str = ""
    login_key: str = ""
    password: str | None = None
    description: str | None = None
